### DISCUSSION TITLE
Undecidability problem.

### DISCUSSION THREAD

**Discussion by A-N-I-K** [Z3Prover/z3#6038] [Q&A]
When solving SMT problems using uninterpreted functions and quantifiers over (bounded) real domains, is there a way to determine whether the solver is in an "undecidable fragment" of SMT solving?

**Comment by NikolajBjorner** [comment]
For uninterpreted domains there is a complete classification: https://web.eecs.umich.edu/~gurevich/Books/CDP.html
With uninterpreted functions and reals - Julia Robinson's work?
